pkg/pipeline/encode/encode_prom.go

```diff
@@ -44,8 +44,10 @@ type PromMetric struct {
 }
 
 type metricInfo struct {
-	input      string
-	labelNames []string
+	input       string
+	filterKey   string
```

Comment:
consider putting key and value in sub-struct  (I just think that will happen if we decide to change the parameters of the filter)